api/extensions/ext_logging.py

```diff
@@ -60,6 +61,46 @@ def time_converter(seconds):
             if handler.formatter:
                 handler.formatter.converter = time_converter
 
+    # for debugging purposes, log the request context
+    @app.before_request
+    def log_request():
+        # only log if DEBUG is enabled
+        if logging.getLogger().isEnabledFor(logging.DEBUG):
+            content_type = flask.request.content_type
+            # only log body if the request is JSON
+            if flask.request.data and "application/json" in content_type.lower():
+                try:
+                    json_data = json.loads(flask.request.data)
+                    formatted_json = json.dumps(json_data, ensure_ascii=False, indent=2)
+                    logging.debug(
```

Comment:
I suggest updating this to the following code:

```py
logging.debug(
    "Received Request %s -> %s, Request Body:\n%s",
    flask.request.method, flask.request.path, formatted_json,
)
```

Using `%` formatting in this context is more efficient than utilizing f-strings. With f-strings, the interpolation occurs immediately when `logging.debug` is called, regardless of whether the message is ultimately logged. In contrast, the suggested approach defers interpolation until the message is actually logged, avoiding unnecessary performance overhead.

For more details, you can refer to this discussion: [StackOverflow](https://stackoverflow.com/questions/11955787/idiomatic-python-logging-format-string-args-list-vs-inline-string-formatting#comment74197244_11955929).